packages/flutter/lib/src/material/range_slider.dart

```diff
@@ -0,0 +1,1446 @@
+      isOnTop: false,
+      sliderTheme: _sliderTheme,
+      thumb: bottomThumb,
+    );
+    _sliderTheme.rangeThumbShape.paint(
+      context,
+      topThumbCenter,
+      activationAnimation: _valueIndicatorAnimation,
+      enableAnimation: _enableAnimation,
+      isDiscrete: isDiscrete,
+      isOnTop: thumbDelta < sliderTheme.rangeThumbShape.getPreferredSize(isEnabled, isDiscrete).width,
+      sliderTheme: _sliderTheme,
+      thumb: _lastThumbSelection,
+    );
+  }
+
+  @override
+  void describeSemanticsConfiguration(SemanticsConfiguration config) {
+    super.describeSemanticsConfiguration(config);
+
+    config.isSemanticBoundary = isEnabled;
+    if (isEnabled) {
+      config.textDirection = textDirection;
+      config.customSemanticsActions = <CustomSemanticsAction, VoidCallback>{
+        _decreaseStart: _decreaseStartAction,
+        _increaseStart: _increaseStartAction,
+        _decreaseEnd: _decreaseEndAction,
+        _increaseEnd: _increaseEndAction,
+      };
+      if (semanticFormatterCallback != null) {
+        config.value = semanticFormatterCallback(_state._lerpRangeValues(values));
+      } else {
+        config.value = values.toString();
+      }
+    }
+  }
+
+  final CustomSemanticsAction _decreaseStart = const CustomSemanticsAction(label: 'Decrease Min');
+  final CustomSemanticsAction _increaseStart = const CustomSemanticsAction(label: 'Increase Min');
+  final CustomSemanticsAction _decreaseEnd = const CustomSemanticsAction(label: 'Decrease Max');
+  final CustomSemanticsAction _increaseEnd = const CustomSemanticsAction(label: 'Increase Max');
+
+  double get _semanticActionUnit =>
+      divisions != null ? 1.0 / divisions : _adjustmentUnit;
+
+  void _increaseStartAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(_increaseValue(values.start), values.end));
+    }
+  }
+
+  void _decreaseStartAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(_decreaseValue(values.start), values.end));
+    }
+  }
+
+  void _increaseEndAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(values.start, _increaseValue(values.end)));
+    }
+  }
+
+  void _decreaseEndAction() {
+    if (isEnabled) {
+      onChanged(RangeValues(values.start, _decreaseValue(values.end)));
+    }
+  }
+
+  double _increaseValue(double value) {
+    return (value + _semanticActionUnit).clamp(0.0, 1.0);
+  }
+
+  double _decreaseValue(double value) {
+    return (value - _semanticActionUnit).clamp(0.0, 1.0);
+  }
+}
+
+/// Decides which thumbs (if any) should be selected.
+///
+/// The default finds the closest thumb, but iff the thumbs are close to each
+/// other, it waits for movement to determine the selected thumb.
+///
+/// Override [RangeSlider.thumbSelector] for custom thumb selection.
+typedef Thumb RangeThumbSelector(
+  TextDirection textDirection,
+  RangeValues values,
+  double tapValue,
+  Size thumbSize,
+  Size trackSize,
+  double dx
+);
+
+/// Object for updating range slider thumb values.
```

Comment:
I think a possible better description of this is:

`Object for representing range slider thumb values`

Since it updates them but is also read by the user to determine what the slider is set to.

Reply:
Done